testing/test_session.py

```diff
@@ -240,6 +240,18 @@ def test_exclude(testdir):
     result.stdout.fnmatch_lines(["*1 passed*"])
 
 
+def test_deselect(testdir):
+    testdir.makepyfile(test_a="""
+        import pytest
+        def test_a1(): pass
+        @pytest.mark.parametrize('b', range(10))
+        def test_a2(b): pass
+    """)
+    result = testdir.runpytest("--deselect=test_a.py::test_a2[1]")
+    assert result.ret == 0
+    result.stdout.fnmatch_lines(["*11 passed, 1 deselected*"])
```

Comment:
This should be `10 passed, 1 deselected` right?

Reply:
Also I think it would be better to use `-v` and deselect two tests and ensure they are not being run. You can parametrize `test_a2` using a smaller value (say 3) so the test remains tidy and small.